web_poet/serialization/api.py

```diff
@@ -0,0 +1,106 @@
+import os
+from functools import singledispatch
+from importlib import import_module
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Optional, Type, TypeVar, Union
+
+import andi
+
+from web_poet import Injectable
+from web_poet.pages import is_injectable
+
+# represents a leaf dependency of any type serialized as a set of files
+SerializedLeafData = Dict[str, bytes]
+# represents a set of leaf dependencies of different types
+SerializedData = Dict[str, SerializedLeafData]
+T = TypeVar("T")
+InjectableT = TypeVar("InjectableT", bound=Injectable)
+SerializeFunction = Callable[[Any], SerializedLeafData]
+DeserializeFunction = Callable[[Type[T], SerializedLeafData], T]
+
+
+def read_serialized_data(directory: Union[str, os.PathLike]) -> SerializedData:
+    result: SerializedData = {}
+    directory = Path(directory)
+    for entry in directory.iterdir():
+        if not entry.is_file():
+            continue
+        if "-" in entry.name:
+            prefix, name = entry.name.split("-", 1)
+        else:
+            prefix, name = entry.name.rsplit(".", 1)
```

Comment:
I think it makes sense to move such logic to a separate function, and have some tests for it; even doctest would do.